packages/circularprogress/src/react/index.tsx

```diff
@@ -80,18 +81,44 @@ const CircularProgress = React.forwardRef<
   CircularProgressProps
 >((props, ref) => {
   const defaultIndeterminateValue = 25
-  const { size, value = defaultIndeterminateValue, ...rest } = props
+  const { size, value, 'aria-label': ariaLabel, ...rest } = props
   const themeName = useTheme()
 
-  const dashOffset = ((100 - value) / 100) * circumference
-
+  const dashOffset =
+    ((100 - (value || defaultIndeterminateValue)) / 100) * circumference
+  const [busy, setBusy] = React.useState<'false' | 'true'>('false')
+  React.useEffect(() => {
+    if (value) {
+      value > 0 && setBusy('true')
+      value === 100 && setBusy('false')
+    }
+    setBusy('true')
+  }, [value])
+  const ariaAttributes = value
+    ? {
+        'aria-label': ariaLabel,
+        'aria-valuemin': 0,
+        'aria-valuemax': 100,
+        'aria-valuenow': Math.round(value)
+      }
+    : {
+        'aria-label': 'Loading'
+      }
   return (
-    <div ref={ref} {...styles.circularprogress(themeName, props)} {...rest}>
-      <ScreenReaderOnly role="region" aria-live="off">
-        {typeof props.value === 'undefined'
-          ? 'Loading...'
-          : `${value}% complete`}
-      </ScreenReaderOnly>
+    <div
+      ref={ref}
+      {...styles.circularprogress(themeName, props)}
+      {...rest}
+      role="progressbar"
+      {...ariaAttributes}
+      aria-busy={busy}
```

Comment:
Reading more about aria-busy, I'm not sure it's what we want here.

> Indicates an element is being modified and that assistive technologies MAY want to wait until the modifications are complete before exposing them to the user.

> The default value of aria-busy is false for all elements. When aria-busy is true for an element, assistive technologies MAY ignore changes to content owned by that element and then process all changes made during the busy period as a single, atomic update when aria-busy becomes false.

https://www.w3.org/TR/wai-aria-1.1/#aria-busy

In the section about progressbar, however, it says aria-busy should be used, but on the (other) region of the page for which loading is being indicated:

> If the progressbar is describing the loading progress of a particular region of a page, the author SHOULD use aria-describedby to point to the status, and set the aria-busy attribute to true on the region until it is finished loading.

https://www.w3.org/TR/wai-aria-1.1/#progressbar

Based on that knowledge, I'd recommend removing it. Do you have other knowledge that says otherwise?

Reply:
I followed Deque on this pattern 🤷🏿‍♂️
https://dequeuniversity.com/library/aria/progress-bar-bounded 
https://dequeuniversity.com/library/aria/progress-bar-unbounded

Reply:
^ Yeah, that's totally not what the ARIA guidelines say. Deque is trusted for a lot, though, so maybe this too.

Practically, if aria-busy can hide reading via screenreader, does removing it solve the other issue I had with not getting good screenreads? If so, venerable though Deque is, I'd still remove it.

This might be a good thing to pass by Amber to get another take.